gpkit/keydict.py

```diff
@@ -116,6 +116,12 @@ def __setitem__(self, key, value):
                 emptyvec = np.full(key.shape, np.nan, **kwargs)
                 dict.__setitem__(self, key, emptyvec)
         for key in self.keymap[key]:
+            if getattr(value, "exp", None) is not None and not value.exp:
```

Comment:
why not `if hasattr(value, "exp") and not value.exp:`? I realize that's not quite equivalent -- is `value.exp` sometimes `None`?